Core.DomainModel/Organization/Organization.cs

```diff
@@ -245,12 +245,27 @@ public Maybe<OperationError> ImportNewExternalOrganizationOrgTree(OrganizationUn
                     organizationUnits.ForEach(unit => unit.ConvertToKitosUnit());
                     StsOrganizationConnection.Disconnect();
                     return new DisconnectOrganizationFromOriginResult(organizationUnits);
-                    break;
                 case OrganizationUnitOrigin.Kitos:
                     return new OperationError("Kitos is not an external source and cannot be disconnected", OperationFailure.BadInput);
                 default:
                     throw new ArgumentOutOfRangeException();
             }
         }
+
+        public Result<OrganizationUnit, OperationError> RemoveOrganizationUnit(int unitId)
+        {
+            var unit = OrgUnits.FirstOrDefault(x => x.Id == unitId);
+            if (unit == null)
+                return new OperationError("Unit doesn't exist in the organization", OperationFailure.NotFound);
+            if (unit.Origin != OrganizationUnitOrigin.Kitos)
+                return new OperationError("Only a KITOS unit can be deleted", OperationFailure.BadState);
+            if (unit.Using.Any() || unit.Rights.Any() || unit.EconomyStreams.Any() ||
```

Comment:
I would prefer that we put a "IsInUse" in the organization unit because that one should be able to answer the question